lib/http.js

```diff
@@ -23,11 +28,22 @@ const client = require('_http_client');
 const ClientRequest = exports.ClientRequest = client.ClientRequest;
 
 exports.request = function request(options, cb) {
+  if (typeof options === 'string') {
+    options = url.parse(options);
+    if (!options.hostname) {
+      throw new Error('Unable to determine the domain name');
+    }
+  } else if (options instanceof url.URL) {
```

Comment:
Actually, it would be best if we could avoid duplicating this kind of `options` check, since it's already being done in `ClientRequest()`. But it would still be nice to avoid `instanceof` in `ClientRequest()`.